### PR TITLE
My hw 04

### PR DESCRIPTION
**Pull request by Shkuryn**
My hw 04 first variant

### CODE REVIEW

File: HW04/ihar_shkuryn/lib/homework.rb [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+  attr_reader :title, :description, :student, :mentor
+  attr_accessor :notices, :answers
+
+  def initialize(data, mentor)
+    @title = data[:title]
+    @description = data[:description]
+    @student = data[:student]
+    @mentor = mentor
+    @status = data[:status]
+    message = "Student #{@student.name} #{@student.surname} got a new homework "\
```
Comment (AliakseiAndryianau) [inline HW04/ihar_shkuryn/lib/homework.rb:13]: consider moving this logic from initializer. it has to serve for assigning values or some very important logic calls

File: HW04/ihar_shkuryn/lib/mentor.rb [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+    @notices = []
+    @all_homeworks = []
+    @homeworks_for_check = []
+  end
+
+  def add_homework(data)
+    homework = Homework.new(data, self)
+    all_homeworks.push(homework)
+    data[:student].homeworks_todo.push(homework)
+    notice = Notification.new(homework.title)
```
Comment (AliakseiAndryianau) [inline HW04/ihar_shkuryn/lib/mentor.rb:20]: move notification logic to a separate method pls

File: HW04/ihar_shkuryn/lib/mentor.rb [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+    homework
+  end
+
+  def subscribe_to!(student)
+    @subscriptions.push(student)
+    student.notices.push("Mentor #{@name} subscribed to #{student.name}")
+  end
+
+  def notifications
+    # any logic here'
```
Comment (AliakseiAndryianau) [inline HW04/ihar_shkuryn/lib/mentor.rb:31]: ?

File: HW04/ihar_shkuryn/lib/student.rb [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+    # any logic here
+  end
+
+  def mark_as_read!
+    notices.clear
+  end
+
+  def homeworks
+    homeworks_todo.each do |hw|
+      # any logic with hw
```
Comment (AliakseiAndryianau) [inline HW04/ihar_shkuryn/lib/student.rb:25]: replace all comment with actual logic please

File: HW04/ihar_shkuryn/lib/homework.rb (hunk 1 above)
Comment (Shkuryn) [reply HW04/ihar_shkuryn/lib/homework.rb]: done

File: HW04/ihar_shkuryn/lib/homework.rb [hunk 5]
```diff
@@ -0,0 +1,15 @@
+# frozen_string_literal: true
+
+class Homework
+  attr_reader :title, :description, :student, :mentor
+  attr_accessor :notices, :answers, :message
+
+  def initialize(data, mentor)
```
Comment (B0RlS) [inline HW04/ihar_shkuryn/lib/homework.rb:7]:
```suggestion
  def initialize(title:, description:, student:, mentor:, status:)
```

File: HW04/ihar_shkuryn/lib/mentor.rb [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+  def initialize(data)
+    @name = data[:name]
+    @surname = data[:surname]
+    @subscriptions = []
+    @notices = []
+    @all_homeworks = []
+    @homeworks_for_check = []
+  end
+
+  def add_homework(data)
```
Comment (B0RlS) [inline HW04/ihar_shkuryn/lib/mentor.rb:16]:
```suggestion
  def add_homework(title:, description:, student:)
```

File: HW04/ihar_shkuryn/lib/mentor.rb [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+    student.notices.push(notice)
+  end
+
+  def subscribe_to!(student)
+    @subscriptions.push(student)
+    student.notices.push("Mentor #{@name} subscribed to #{student.name}")
+  end
+
+  def notifications
+    @notices.each do |notice|
```
Comment (B0RlS) [inline HW04/ihar_shkuryn/lib/mentor.rb:38]:
```suggestion
    notices.each do |notice|
```

File: HW04/ihar_shkuryn/lib/mentor.rb [hunk 8]
```diff
@@ -0,0 +1,58 @@
...
+    homework
+  end
+
+  def notice_handler(homework, student)
+    notice = Notification.new(homework.title)
+    student.notices.push(notice)
+  end
+
+  def subscribe_to!(student)
+    @subscriptions.push(student)
```
Comment (B0RlS) [inline HW04/ihar_shkuryn/lib/mentor.rb:33]:
```suggestion
    subscriptions.push(student)
```
add subscriptions to attr_reader

File: HW04/ihar_shkuryn/lib/mentor.rb [hunk 9]
```diff
@@ -0,0 +1,58 @@
...
+    student.notices.push("Mentor #{@name} subscribed to #{student.name}")
+  end
+
+  def notifications
+    @notices.each do |notice|
+      puts notice
```
Comment (B0RlS) [inline HW04/ihar_shkuryn/lib/mentor.rb:39]:
```suggestion
     notice
```

**Pull request merged**